### PR TITLE
feature: support tencent polaris discovery 

### PR DESCRIPTION
**Pull request by misselvexu**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
- Add Tencent Polaris Registry (Configuration & Service Discovery) support for Seata framework.

### Ⅱ. Does this pull request fix one issue?

#4867.

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
None.

### Ⅳ. Describe how to verify it
**A question**: How is the Seata framework implemented when doing integration testing ?

### Ⅴ. Special notes for reviews
None.

### CODE REVIEW

**Comment by slievrly** [comment]
@misselvexu Thank you very much for your contribution, I will review this PR later.

**Comment by slievrly** [comment]
add configuration item on follow files(used for client):
https://github.com/seata/seata/tree/develop/script/client/spring

**Comment by misselvexu** [comment]
@slievrly  How should I conduct integration testing? Is there any relevant case ?

**Comment by misselvexu** [comment]
> some check style not pass:
> image
> 

i will fix this issue later。

**Comment by slievrly** [comment]
> @slievrly How should I conduct integration testing? Is there any relevant case ?

Currently, there is no integration test related to registration and configuration center. `seata-test` module is used for integration testing, but this module will be reconstructed into integration testing based on testcontainers. At present, I think we can conduct mock test based on the third-party component itself.

File: config/seata-config-polaris/src/main/java/io/seata/config/polaris/client/SimpleHttpRequest.java [hunk 1]
```diff
@@ -0,0 +1,2996 @@
...
+                        }
+
+                        public void checkServerTrusted(X509Certificate[] chain, String authType) {
+                            // Intentionally left blank
+                        }
+                    }
+                };
+            try {
+                SSLContext context = SSLContext.getInstance("TLS");
+                context.init(null, trustAllCerts, new SecureRandom());
```
[the remark this answers is not in the document]
Comment (misselvexu) [reply config/seata-config-polaris/src/main/java/io/seata/config/polaris/client/SimpleHttpRequest.java]: fixed.

File: discovery/seata-discovery-polaris/src/main/java/io/seata/discovery/registry/polaris/client/SimpleHttpRequest.java [hunk 2]
```diff
@@ -0,0 +1,2995 @@
...
+                        }
+
+                        public void checkServerTrusted(X509Certificate[] chain, String authType) {
+                            // Intentionally left blank
+                        }
+                    }
+                };
+            try {
+                SSLContext context = SSLContext.getInstance("TLS");
+                context.init(null, trustAllCerts, new SecureRandom());
```
[the remark this answers is not in the document]
Comment (misselvexu) [reply discovery/seata-discovery-polaris/src/main/java/io/seata/discovery/registry/polaris/client/SimpleHttpRequest.java]: fixed.

**Comment by opcooc** [comment]
嗨 大概多久可以合并 很期待

**Comment by a364176773** [comment]
> 嗨 大概多久可以合并 很期待

1.6发版后开始review

**Comment by opcooc** [comment]
> > 嗨 大概多久可以合并 很期待
> 
> 1.6发版后开始review

好的

**Comment by JavaLionLi** [comment]
2.0.0没有合并吗 review进展到什么程度了啊

**Comment by funky-eyes** [comment]
Due to code conflicts and age, if you are still interested in participating in the community, please resubmit this pr to the 2.x branch after resolving the conflict.
由于代码冲突以及年代久远，如果你还有兴趣参与到社区，请解决完冲突后重新请提交这份pr至2.x分支中

**Comment by misselvexu** [comment]
> Due to code conflicts and age, if you are still interested in participating in the community, please resubmit this pr to the 2.x branch after resolving the conflict. 由于代码冲突以及年代久远，如果你还有兴趣参与到社区，请解决完冲突后重新请提交这份pr至2.x分支中

After optimize the latest code, am I reapplying to re-open this PR?

**Pull request closed without merging**